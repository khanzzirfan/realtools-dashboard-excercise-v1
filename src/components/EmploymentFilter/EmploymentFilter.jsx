import styled from 'styled-components';

export const EmploymentGroup = styled.div`
  display: grid;
  grid-template-columns: repeat(2, auto);
  align-items: center;
  column-gap: 20px;
`;
