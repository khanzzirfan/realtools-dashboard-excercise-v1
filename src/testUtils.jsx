import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MemoryRouter } from 'react-router-dom';
import { AppContextProvider } from './hooks/useAppContext';
import { theme } from './theme';

const AllTheProviders = ({ children }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        // âœ… turns retries off
        retry: 2,
      },
    },
  });
  return (
    <QueryClientProvider client={queryClient}>
      <AppContextProvider>
        <ThemeProvider theme={theme}>
          <MemoryRouter initialEntries={['/people', '/people/new', '/people/edit/:id']}>
            {children}
          </MemoryRouter>
        </ThemeProvider>
      </AppContextProvider>
    </QueryClientProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
