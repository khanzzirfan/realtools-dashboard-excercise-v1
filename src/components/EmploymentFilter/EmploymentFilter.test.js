import '@testing-library/jest-dom';
import { render } from '../../testUtils';
import { EmploymentGroup } from './EmploymentFilter';

describe('EmploymentGroup', () => {
  it('should renders correctly with custom css', () => {
    const { asFragment } = render(<EmploymentGroup>Contractor</EmploymentGroup>);
    expect(asFragment()).toMatchSnapshot();
  });
});
