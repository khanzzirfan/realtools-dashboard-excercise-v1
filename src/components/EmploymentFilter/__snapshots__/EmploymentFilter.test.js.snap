// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmploymentGroup should renders correctly with custom css 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-template-columns: repeat(2,auto);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 20px;
  column-gap: 20px;
}

<div
    class="c0"
  >
    Contractor
  </div>
</DocumentFragment>
`;
